### ISSUE TITLE
[BUG] OpenSearch version of core only

### ISSUE THREAD

**Issue by stockholmux** [opensearch-project/OpenSearch#1111]
**Describe the bug**
When a user connects to `https://<ip-of-cluster>:9200/` the response JSON reports `version.number` for the core (e.g. 1.0.0). Looking at the plans for 1.0.1 release, it looks like only particular bundled plugins will be updated and not the core. This means if a users updates to version 1.0.1, `https://<ip-of-cluster>:9200/` will still respond with 1.0.0 in `version.number`. This could cause confusion after updates or if a user really needs to distinguish between versions.

**To Reproduce**
1. Go to `curl https://localhost:9200/ -u admin:admin -k`
2. See the response - something like:
```
{
  "name" : "opensearch-node1",
  "cluster_name" : "opensearch-cluster",
  "cluster_uuid" : "8oUPWeGFSTOH5cwKaBfuwA",
  "version" : {
    "distribution" : "opensearch",
    "number" : "1.0.0",
    "build_type" : "tar",
    "build_hash" : "34550c5b17124ddc59458ef774f6b43a086522e3",
    "build_date" : "2021-07-02T23:22:21.383695Z",
    "build_snapshot" : false,
    "lucene_version" : "8.8.2",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}

**Expected behaviour**
The user should be able to see the bundled version (as well as the core version) when hitting `https://<ip-of-cluster>:9200/`

Perhaps something like this:
```
{
  "name" : "opensearch-node1",
  "cluster_name" : "opensearch-cluster",
  "cluster_uuid" : "8oUPWeGFSTOH5cwKaBfuwA",
  "version" : {
    "bundle" : "1.0.1"
    "distribution" : "opensearch",
    "number" : "1.0.0",
    "build_type" : "tar",
    "build_hash" : "34550c5b17124ddc59458ef774f6b43a086522e3",
    "build_date" : "2021-07-02T23:22:21.383695Z",
    "build_snapshot" : false,
    "lucene_version" : "8.8.2",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
```
(see `version`.`bundle`)

Although, I could see a different endpoint for the bundle version...



**Plugins**
All OpenSearch

**Comment by dblock** [comment]
@nknize We have a mechanism in 1.1 to describe the bundle, a bundle manifest that [looks like this](https://github.com/opensearch-project/opensearch-build/blob/main/bundle-workflow/tests/test_manifests/data/opensearch-bundle-1.1.0.yml). It's going to be dropped along with the rest of the world in the bundle and we could read from that.

So we have two choices:

1. In a bundle installation, change version.number=bundle version, build_type=tar, build_hash=some kind of bundle hash, build_date=the date of the bundle build from the bundle JSON. 
2. Return the bundle versioning data somewhere else if present. Where?

(1), from the machine point of view this, at first, sounded like a bad idea, because you could get two different versions for the same `-min` engine.  But bundle closely follows `-min` in versioning, so semantically it would still be correct. So feels like not a big deal, to me.

WDYT?

**Comment by nknize** [comment]
> I guess the line between a bug and an enhancement to me is "would the average user be surprised by a behaviour of the software."

I agree with this. If a user has a bad time it's a bug. That's a valid point! Maybe we create a new github label `user_experience` to capture issues like this vs runtime or implementation bugs? Might be a good way for users to gain merit through design contributions? I wonder if other projects have something like this? /cc @spotaws 



> It's going to be dropped along with the rest of the world in the bundle and we could read from that.

I think this is a good choice if we have a new plugin called "bundle manager" or something? This plugin could expose an API for getting all details about the bundle rather than modifying core OpenSearch with logic about the bundle (decouple from the core). I do have some questions about this, though:

1. What is a bundle? A list of "blessed" plugins the project deems as first class?
2. Bundle plugins are installed by default? But users can uninstall them if they don't want them? 
3. Does the "org" create a process for incubating third-party plugins with a well defined path to become a first class bundle plugin?
4. Instead of a single blessed bundle, what about creating many bundles consisting of groups of similar plugins? (e.g., ML bundle, Spatial bundle, Observability bundle)? In this manner there's one opensearch w/ many experience options?
5. Until we have a well defined version design, the many to one versioning is going to be an issue. How is it defined? 

I have some more but I'll stick with these four for now...

**Comment by stockholmux** [comment]
I agree that a nice vision would be some sort of "choose your own adventure" OpenSearch, but that opens up a lot of considerations (dependency management/resolution, etc.).

From my personal POV, the bundled plugins are basically like a standard library in a programming language. Perhaps there are lessons in process & versioning from that world? I think that might inform some responses to @nknize's questions.

**Comment by stockholmux** [comment]
Is this fixed for the upcoming 1.1.1 release?

**Comment by dagneyb** [comment]
@dblock did this get fixed?

**Comment by dblock** [comment]
@dagneyb Nope. It's really a new feature where we want to repor the version of the whole distribution.
